docs/reference/settings.md

```diff
@@ -269,6 +269,45 @@ package = false
 
 ---
 
+### [`required-environments`](#required-environments) {: #required-environments }
+
+A list of required platforms, for packages that lack source distributions.
+
+When a package does not have a source distribution, it's availability will be limited to
+the platforms supported by its built distributions (wheels). For example, if a package only
+publishes wheels for Linux, then it won't be installable on macOS or Windows.
+
+By default, uv requires each package to include at least one wheel that is compatible with
+the designated Python version. The `required-environments` setting can be used to ensure that
+the resulting resolution contains wheels for specific platforms, or fails if no such wheels
+are available.
```

Comment:
I would add something about using older versions here, like "If a version of a package does not contain the required source distribution or wheels, uv will try older versions until it finds a version with sufficient support".